### PR TITLE
Include Fly infra domains in the allowed domains list for synthetic checks

### PR DESCRIPTION
**Pull request by aschiavo**
### Change Summary

What and Why:

Update the target check to include all Fly infra domains

### CODE REVIEW

File: internal/metrics/synthetics/agent.go [hunk 1]
```diff
@@ -139,7 +139,11 @@ func isFlyInfraTarget(target string) bool {
 		host = parsedURL.Host
 	}
 
-	_, existsInFlyInfraDomains := flyInfraHosts[host]
+	for _, flyInfraDomain := range flyInfraDomains {
+		if strings.HasSuffix(host, "."+flyInfraDomain) || host == flyInfraDomain {
```
Comment (davidgraig) [inline internal/metrics/synthetics/agent.go:143]: Could just return the evaluation here I think?

**Review by davidgraig — APPROVED** [review]
Lgtm

File: internal/metrics/synthetics/agent.go (hunk 1 above)
Comment (aschiavo) [reply internal/metrics/synthetics/agent.go:143]: Hmmm, not sure I fully get the comment 😅 I _think_ I am . If the target host ends with any of the allowed domains or is the naked domain itself I'm returning the evaluation as positive, if the loop ends and it haven't returned for any of the allowed domains I return a nope ... What am I missing 🤔

Comment (davidgraig) [reply internal/metrics/synthetics/agent.go:143]: oh. nevermind i was looking at this on my phone. NEVERMIND ME.

**Pull request merged**